Jenkinsfile

```diff
@@ -92,6 +92,20 @@ echo ${libs} | sed -e 's/,/ /g' | xargs md5sum
 """
 }
 
+def collect_test_results_unix(original_file_name, new_file_name) {
+    echo 'Saving python test results for ' + new_file_name
+    // Rename file to make it distinguishable. Unfortunately, it's not possible to get STAGE_NAME in a parallel stage
+    sh 'cp ' + original_file_name + ' ' + new_file_name
+    archiveArtifacts artifacts: new_file_name
+}
+
+def collect_test_results_windows(original_file_name, new_file_name) {
```

Comment:
Why do you copy first?

Reply:
The copy is in order to have the file with a unique filename. archiveArtifacts identifies files by their filename. If we have two scripts producing a file called 'nosetests_python_unittest.xml', it will override the first with the second. This copy ensures that the files gets renamed to a unique-per-run filename.